src/Fantomas/Context.fs

```diff
@@ -510,6 +510,9 @@ let internal indentOnWith (ctx : Context) =
 /// Conditional unindentation on with keyword
 let internal unindentOnWith (ctx : Context) =
     if ctx.Config.IndentOnTryWith then unindent ctx else ctx
+    
+let internal sepBetweenModuleAndChildren (ctx : Context) =
+    if ctx.Config.EmptyLineBeforeNestedModuleBody then (rep 2 sepNln) ctx else sepNln ctx
```

Comment:
You might need `sepNlnConsideringTriviaContentBefore` or `sepNlnConsideringTriviaContentBeforeWithAttributes` instead of `rep 2 sepNln`. Do you have a unit test where you already have the extra empty line in the source code?
I would expect that you end up with two extra empty line with this code.
What happens if you format this code?

```fsharp
module x =

   let y = 8
```

and

```fsharp
module x =

    [<Foo>]
    let y = 8
```